lib/acts-as-taggable-array-on/taggable.rb

```diff
@@ -2,14 +2,24 @@
 
 module ActsAsTaggableArrayOn
   module Taggable
+    class TaggableError < StandardError; end
+
+    class InvalidAllowListTypeError < TaggableError
+      def to_s
+        "Allow list has to be an array"
+      end
+    end
+
     def self.included(base)
       base.extend(ClassMethod)
     end
 
     TYPE_MATCHER = {string: "varchar", text: "text", integer: "integer"}
 
     module ClassMethod
-      def acts_as_taggable_array_on(tag_name, *)
+      def acts_as_taggable_array_on(tag_name, **args)
+        raise InvalidAllowListTypeError if args[:allow_list].present? && !args[:allow_list].is_a?(Array)
```

Comment:
Really like your proposal, I personally don't feel the need for this functionality right now, but can easily see how this could be useful! 

But I feel reluctant to merge proposed implementation due to this error throwing. ActiveRecord by default doesn't throw errors, it would seem unnatural if we would do it. 

Instead most common pattern is that we check that object is `.valid?` and if we want to get more details about errors we would just retrieve all of them with `.errors`. Only methods with a bang in the end, change this behavior.

Can I suggest, that you drop this error and not throw anything? Would you agree that this seems more "activerecord" way?